nixos/modules/services/networking/xray.nix

```diff
@@ -80,9 +80,12 @@ with lib;
       description = "xray Daemon";
       after = [ "network.target" ];
       wantedBy = [ "multi-user.target" ];
+      script = ''
+        exec "${cfg.package}/bin/xray" -config "${CREDENTIALS_DIRECTORY}/config.json"
```

Comment:
Your code is invalid, since `${CRE...}` is not the same as `$CRE...`
The right code is:
`exec "${cfg.package}/bin/xray" -config "$CREDENTIALS_DIRECTORY/config.json"`

Reply:
Ok, now I see the difference it makes. Fixed up